examples/scripts/TreeCompareAuto.C

```diff
@@ -0,0 +1,293 @@
+
+        if (*htemp == *hist1) {
+            numIdenticalHistograms++;
+            outstream << "Histograms are identical." << std::endl;
+            outstream << "**********************" << std::endl;
+            hist1->Clear();
+            hist->Clear();
+            htemp->Clear();
+            continue;
+        }
+
+        // if also the histograms are not identical check if the histograms
+        // are equal on a statistical base. Use The Kolmogorov test for
+        // this.
+        outstream << "Checking Kolmogorov" << endl;
+
+        double kolmo = htemp->KolmogorovTest(hist1);
+
+        outstream << "Result of Kolmogorov test: " << kolmo << endl;
+        if (kolmo > 0.99) {
+            numKolmogorovHistograms++;
+            outstream << "**********************" << std::endl;
+            hist1->Clear();
+            hist->Clear();
+            htemp->Clear();
+            continue;
+        }
+
+        outstream << "Data are differnt" << std::endl;
+        outstream << "**********************" << std::endl;
+        outstream << "Entries: " << hist->GetEntries() << std::endl;
+        outstream << "Mean: " << hist->GetMean() << std::endl;
+        outstream << "RMS: " << hist->GetRMS() << std::endl;
+        outstream << "Underflow: " << hist->GetBinContent(0) << std::endl;
+        outstream << "Overflow: " << hist->GetBinContent(hist->GetNbinsX() + 1) << std::endl;
+        outstream << "**********************" << std::endl;
+        okay = false;
+        numFailedLeaves++;
+        hist1->Clear();
+        hist->Clear();
+        htemp->Clear();
+    }
+
+    if (!okay) {
+        std::cout << outstream.str();
+        std::cout << "Test failed." << std::endl;
+        std::cout << numFailedLeaves << " of " << numTestedLeaves << " leaves are different." << std::endl;
+        return 1;
+    }
+
+    //  std::cout << outstream.str();
+    std::cout << "Tested leaves:                    " << numTestedLeaves << std::endl;
+    //  std::cout << "Empty leaves:                     " << numEmptyLeaves << std::endl;
+    std::cout << "Leaves with identical entries:    " << numIdenticalEntries << std::endl;
+    std::cout << "Leaves with identical histograms: " << numIdenticalHistograms << std::endl;
+    std::cout << "Leaves with kolmo histograms:     " << numKolmogorovHistograms << std::endl;
+    std::cout << "Test passed. All leaves of all branches are exactly identical." << std::endl;
+
+    return 0;
+}
+
+bool operator==(TH1 const& lhs, TH1 const& rhs)
+{
+    for (int x = 0; x < lhs.GetNbinsX() + 1; ++x) {
+        if (lhs.GetBinContent(x) != rhs.GetBinContent(x)) {
+            return false;
+        }
+    }
+    return true;
+}
+
+std::vector<TString> GetLeafNames(TTree& cbmsim)
+{
+
+    std::vector<TString> ListOfLeaves;
+
+    TObjArray* bla1 = cbmsim.GetListOfLeaves();
+    TIter myiter1(bla1);
+    TBranch* branch;
+    while ((branch = (TBranch*)myiter1.Next())) {
```

Comment:
Here you actually have a classic fail with C-style cast. This container does not contain `TBranch` objects at all, but `TLeaf`s ... The only reason this went unnoticed, is that you use an interface of a common base class `TNamed`!

Reply:
You are right. Already the function name should have told me that.